pkg/p2p/kadcast/utils.go

```diff
@@ -0,0 +1,110 @@
+package kadcast
+
+import (
+	"encoding/binary"
+	"errors"
+	"log"
+	"net"
+
+	"golang.org/x/crypto/sha3"
+
+	// Just for debugging purposes
+	_ "fmt"
+)
+
+// ------------------ DISTANCE UTILS ------------------ //
+// Computes the XOR distance between 2 different
+// ids.
+func idXor(a *[16]byte, b *[16]byte) uint16 {
+	distance := [16]byte{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}
+	i := 0
+
+	for i < 16 {
+		distance[i] = a[i] ^ b[i]
+		i++
+	}
+	return classifyDistance(&distance)
+}
+
+// This function gets the XOR distance as a byte-array
+// and collapses it to classify the distance on one of the
+// 128 buckets.
+func classifyDistance(arr *[16]byte) uint16 {
```

Comment:
Same for this function. Looks like you don't need to modify `arr`.

Reply:
Same as avobe!